### DISCUSSION TITLE
[BUG]  Using BokehJS as a javascript module: Uncaught syntax error @bokeh.slickgrid.v2.4.4102.js does not provide an export named 'Data'

### DISCUSSION THREAD

**Discussion by RT-Tap** [bokeh/bokeh#12016] [General]
### Software versions

Because I am only interested in the BokehJS module I will list the relevent information, if more is needed please let me know.

Node.js: v16.13.1
npm: 8.1.2
@Bokeh/bokehjs: 2.4.2
Operating System: Windows 10 Home 64 bit (10.0, Build 19042)
Package bundling: Snowpack 3.3.7

### Browser name and version

Firefox, Brave/Chrome

### Expected behavior

I expected BokehJS to import correctly by using `import * as Bokeh from '@bokeh/bokehjs'` in my JS file but instead the only way I can get anything to import/work is by using the includes i;n my html file
```
    <script type="text/javascript" src="https://cdn.bokeh.org/bokeh/release/bokeh-2.4.2.min.js"></script>
    <script type="text/javascript" src="https://cdn.bokeh.org/bokeh/release/bokeh-gl-2.4.2.min.js"></script>
    <script type="text/javascript" src="https://cdn.bokeh.org/bokeh/release/bokeh-widgets-2.4.2.min.js"></script>
    <script type="text/javascript" src="https://cdn.bokeh.org/bokeh/release/bokeh-tables-2.4.2.min.js"></script>
    <script type="text/javascript" src="https://cdn.bokeh.org/bokeh/release/bokeh-mathjax-2.4.2.min.js"></script>
    <script type="text/javascript" src="https://cdn.bokeh.org/bokeh/release/bokeh-api-2.4.2.min.js"></script>
    <script type="text/javascript" src="https://cdn.bokeh.org/bokeh/release/bokeh-api-2.4.2.min.js"></script>
```

### Observed behavior

I get the following error 
```
Uncaught SyntaxError: The requested module '/_snowpack/pkg/@bokeh.slickgrid.v2.4.4102.js' does not provide an export named 'Data'
```

### Example code
initialize environment 
```Python
$ npm init
$ npm i snowpack
$ npm i @bokeh/bokehjs
```

#### index.html
```
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
    <title>title</title>
  </head>
  <body>
    <script type="module" src="app.js"></script>
  </body>
</html>
```

#### app.js:
```
import * as Bokeh from '@bokeh/bokehjs'
```
start a development server
```Python
$ snowpack dev   
```

### Stack traceback or browser console output

Developer console in browser:

```bokeh-05686c2e.js:16 Uncaught SyntaxError: The requested module '/_snowpack/pkg/@bokeh.slickgrid.v2.4.4102.js' does not provide an export named 'Data'```

### Screenshots
![bokehissue](https://user-images.githubusercontent.com/71105781/154818503-36365806-5da4-44ef-aa02-a7c12b6f8e73.jpg)

![bokehissue2](https://user-images.githubusercontent.com/71105781/154818491-9b0b15d2-9d00-4066-a113-c95cec00e582.jpg)

**Comment by mattpap** [comment]
> `Uncaught SyntaxError: The requested module '/_snowpack/pkg/@bokeh.slickgrid.v2.4.4102.js' does not provide an export named 'Data'`

`@bokeh/slickgrid` does export `Data` [here](https://github.com/bokeh/SlickGrid/blob/c4caca521385f62e355dcc40bb4fdd0a145d7b26/index.js#L5). Given this is a `SyntaxError`, I'm guessing that snowpack doesn't understand this style of exports and expects ESM import/export. This may require configuration on your side. Alternatively our fork of slickgrid could be refactored into a proper set of ESM modules.

**Reply by RT-Tap** [reply]
Yea, so I did a lot more searching and I had a feeling that was the cause, unfortunately I don't have a firm grasp on snowpack or packing utilities in general so I am still trying to figure out how to configure snowpack to accommodate this, if I figure it out I will report back. Thank you for your help.
